numpy/_core/src/multiarray/dlpack.c

```diff
@@ -189,33 +249,17 @@ array_dlpack(PyArrayObject *self,
             PyErr_SetString(PyExc_BufferError,
                     "DLPack only supports IEEE floating point types "
                     "without padding (longdouble typically is not IEEE).");
-            return NULL;
+            return -1;
         }
         managed_dtype.code = kDLComplex;
     }
     else {
         PyErr_SetString(PyExc_BufferError,
                 "DLPack only supports signed/unsigned integers, float "
                 "and complex dtypes.");
-        return NULL;
-    }
-
-    DLDevice device = array_get_dl_device(self);
-    if (PyErr_Occurred()) {
-        return NULL;
-    }
-
-    // ensure alignment
-    int offset = sizeof(DLManagedTensor) % sizeof(void *);
```

Comment:
This offset calculation was just wrong.  It simply didn't matter because the struct size is a multiple of this anyway in practice.